src/main/java/com/nirima/jenkins/plugins/docker/DockerCloud.java

```diff
@@ -361,26 +365,32 @@ int countContainersInProgress() {
                         t.getImage(), label, getDisplayName());
 
                 final CompletableFuture<Node> plannedNode = new CompletableFuture<>();
-                r.add(new NodeProvisioner.PlannedNode(t.getDisplayName(), plannedNode, t.getNumExecutors()));
+                DockerNodeFactory.DockerPlannedNode pn =
+                        NODE_FACTORY.createPlannedNode(t.getDisplayName(), plannedNode, t.getNumExecutors(),
+                                getDisplayName(), t.getImage(), label.getName());
 
+                r.add(pn);
                 final Runnable taskToCreateNewSlave = new Runnable() {
                     @Override
                     public void run() {
                         DockerTransientNode slave = null;
                         try {
-                            // TODO where can we log provisioning progress ?
+                            // TODO where can we log provisioning progress ? // In the CloudMonitor API!
+                            pn.notifyStarted(); // get().onStarted(id);
                             final DockerAPI api = DockerCloud.this.getDockerApi();
                             slave = t.provisionNode(api, TaskListener.NULL);
                             slave.setDockerAPI(api);
                             slave.setCloudId(DockerCloud.this.name);
-                            plannedNode.complete(slave);
+                            pn.notifySuccess(slave); //get().onComplete(id, slave);
```

Comment:
As above; either have a human comment or no comment; commented-out code looks like it's a work in progress that needs fixing.